src/app/interceptors/custom-header.interceptor.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ *    Copyright 2019 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { version } from '../../../package.json';
+
+/**
+ * An interceptor that ensures that every request has a custom X-Dockstore-UI header with the UI version as
+ * the value so we can track API calls coming from the Dockstore UI.
+ */
+@Injectable()
+export class CustomHeaderInterceptor implements HttpInterceptor {
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    req = req.clone({ headers: req.headers.set('X-Dockstore-UI', version) });
```

Comment:
I don't think we keep the package.json version up to date, although we should. Right now it's at 2.3.0 in the package.json, but our UI version is 2.5.1. Can you get the version from the same place the footer gets it instead?

Reply:
The footer gets it from ``dockstore-ui/src/app/footer/versions.ts``, but I don't see a UI version there, just the webservice version. The webservice version is set in ``git.version.ts`` using ``process.env.npm_package_config_webservice_version``, but I'm not sure where this is set in order to do a similar thing for the ui version

Reply:
Ignore my previous comment, I misunderstood the file. The UI version is there